genny/auth/templates/models/user.go.plush

```diff
@@ -31,15 +35,48 @@ type User struct {
 // Create wraps up the pattern of encrypting the password and
 // running validations. Useful when writing tests.
 func (u *User) Create(tx *pop.Connection) (*validate.Errors, error) {
-	u.Email = strings.ToLower(u.Email)
-	ph, err := bcrypt.GenerateFromPassword([]byte(u.Password), bcrypt.DefaultCost)
+	verrs := u.validatePassword()
+	if verrs.HasAny() {
+		return verrs, nil
+	}
+
+	ph, err := encryptPassword(u.Password)
 	if err != nil {
 		return validate.NewErrors(), errors.WithStack(err)
 	}
 	u.PasswordHash = string(ph)
+	u = u.sanitizeFields()
 	return tx.ValidateAndCreate(u)
 }
 
+// Update handles the extra work possibly needed during user update,
+// hashing password and making email lowercase for consistency
+func (u *User) Update(tx *pop.Connection) (*validate.Errors, error) {
+	if len(u.Password) != 0 {
+		verrs := u.validatePassword()
+		if verrs.HasAny() {
+			return verrs, nil
+		}
+		ph, err := encryptPassword(u.Password)
+		if err != nil {
+			return validate.NewErrors(), errors.WithStack(err)
+		}
+		u.PasswordHash = ph
+	}
+	u = u.sanitizeFields()
+	return tx.ValidateAndUpdate(u)
+}
+
+func (u *User) sanitizeFields() *User {
+	// force email to lowercase for better matching
+	u.Email = strings.ToLower(u.Email)
```

Comment:
In theory, email addresses are case sensitive according to RFC 5321:
> The local-part of a mailbox MUST BE treated as case sensitive.

In praxis, this should not really matter, but should be documented. A hint at it is existent in the docs for `Update` but not for `Create`, I also added an comment over there